packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -204,19 +236,33 @@ const fetchAggregationData = (page: number): ThunkAction<
       maxTimeMS,
       collation,
       dataService: { dataService },
-      aggregation: { limit, documents: _documents, page: _page, isLast: _isLast },
+      aggregation: {
+        limit,
+        page: _page,
+        isLast: _isLast,
+        documents: _documents,
+        abortController: _abortController,
+      },
     } = getState();
 
     if (!dataService) {
       return;
     }
 
+    // Cancel the existing aggregate
+    _abortAggregation(_abortController);
+
     try {
       const abortController = new AbortController();
       const signal = abortController.signal;
       dispatch({
         type: ActionTypes.AggregationStarted,
         abortController,
+        previousPageData: {
+          page: _page,
+          isLast: _isLast,
+          documents: _documents,
+        },
```

Comment:
Actions should not be the ones deriving new state from current one, that's the responsibility of the reducer